File: src/sequence_metrics/run_motif_search.jl
using Pkg

cd(@__DIR__)
# TODO create julia project in tps_eval
# Pkg.activate(".") # TODO uncomment
Pkg.activate("../../../../terpene_generation/src") # TODO delete
#Pkg.instantiate()

# --- Script ---------------------------------------------------------
include("motif_search.jl")

fasta_path = ARGS[1] # Path to the FASTA file with sequences
motifs = ARGS[2:end] # Motifs to search for, passed as command line arguments
motif_search(fasta_path, motifs; save=true)

File: src/sequence_metrics/run_motif_search.py
from __future__ import annotations

import sys

from motif_search import motif_search


def main() -> None:
    fasta_path = sys.argv[1]
    motifs = sys.argv[2:]
    motif_search(fasta_path, motifs, save=True)


if __name__ == "__main__":
    main()
